### DISCUSSION TITLE
"    - |+ " and "    - |" what is the difference

### DISCUSSION THREAD

**Discussion by hktalent** [projectdiscovery/nuclei#1876] [Q&A] [answered]
"    - |+ " and "    - |" what is the difference

thanks

**Comment by hktalent** [comment]
https://nuclei.projectdiscovery.io/template-examples/http-smuggling/
My understanding is
Wrong agreement when executed
Content-Length: 6
wait for correction

**Comment by ehsandeep** [comment]
@hktalent ` - |+` is required to use with [unsafe HTTP templates](https://nuclei.projectdiscovery.io/templating-guide/protocols/http/#unsafe-http-requests), i.e `unsafe: true`.

**Reply by hktalent** [reply]
ok, thanks for you

**Comment by forgedhallpass** [comment, marked as answer]
@hktalent 

They are standard YAML syntax for working with multi-line strings.

## Block scalars
### Block style indicators
Indicates how newlines inside the block should behave
* `|` (**literal** style): new lines are kept as new lines
* `>` (**folded** style): new lines are replaced by spaces
### Block chomping indicators
Controls what should happen with new lines at the end of a string
* Default is no value (**clip**): adds a single new line at the end of the string
* `-` (**strip**): removes all new line characters from the end of the string
* `+` (**keep**): keeps all new lines at the end

**TL;DR**
> " - |+ " and " - |" what is the difference

`|+`: keep all manually inserted new lines at the end of the string block
`|`: only one new line character is returned at the end of the string block

For more information, visit: https://yaml-multiline.info/

**Reply by hktalent** [reply]
@forgedhallpass thank you my friend
